manifests/init.pp

```diff
@@ -258,6 +262,7 @@
   Optional[String]                     $ldap_base             = undef,
   Optional[String]                     $ldap_host             = undef,
   Optional[String]                     $ldap_options          = undef,
+  String                               $lookup_table_type     = 'hash',
```

Comment:
Hum :thinking: 

This single setting change multiple mappings.  Maybe we should rename this something like `default_lookup_table_type`, and for each mapping have a local type which defaults to `postfix::default_lookup_table_type`.  So same end-result by default but the possibility to use two different types for transport and virtual maps for example?

Reply:
Well, i don't think that is required. 99,9% most probably will continue to use "hash" and don't care about the table type at all, at least until today no interest was shown here to have it more flexible. postfix::hash is used in several places, postfix::map as well, it will get very complex to offer a configurable different table type for each and every purpose.
Most people will probably want to change the general type for everything to a different one (like me f.e. for SLES 15 to lmdb). People with very special postfix setups will construct their setup anyway on their own with the postfix::* defines and don't use the predefined ones like "mta".
My two cents...

Reply:
Good point.  I  am fine with it!